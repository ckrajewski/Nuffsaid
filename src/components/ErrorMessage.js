import React, { useState, useEffect } from 'react';
import { IconButton, Grid, Paper } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  paper: {
    paddingLeft: '3%',
    paddingRight: '8%',
    height: '3em',
    backgroundColor: '#F56236',
  },
  container: {
    width: '40%',
    margin: 'auto',
    marginBottom: '0.25%',
  },
  message: {
    paddingTop: '3%',
  },
  clear: {
    alignSelf: 'center',
  },
  grid: {
    height: '100%',
  },
});

export default function ErrorMessage(props) {
  const [errorObj, setErrorObj] = useState({ timeOutId: null, errorMessage: null });
  const { errorMessage, timeOutId } = errorObj;
  const { message } = props;
  const clearMessage = (event) => {
    event.stopPropagation();
    setErrorObj({ timeOutId: null, errorMessage: null });
  };
  const showErrorForTwoSeconds = () => (setTimeout(() => {
    setErrorObj({ timeOutId: null, errorMessage: null });
  }, 2000));
  useEffect(() => {
    if (message) {
      const { priority } = message;
      if (priority === 1) {
        if (timeOutId) {
          clearTimeout(timeOutId);
        }
        setErrorObj({ errorMessage: message.message, timeOutId: showErrorForTwoSeconds() });
      }
    }
  }, [message]);

  const classes = useStyles();

  return (
    <div className={classes.container} style={{ visibility: errorMessage ? 'initial' : 'hidden' }}>
      <Paper
        elevation={5}
        className={classes.paper}>
        <Grid
          container
          direction="row"
          alignItems="center"
          className={classes.grid}>
          <Grid item>
            <IconButton onClick={clearMessage}>
              <CloseIcon />
            </IconButton>
          </Grid>
          <Grid item>
            {errorMessage}
          </Grid>
        </Grid>
      </Paper>
    </div>
  );
}
