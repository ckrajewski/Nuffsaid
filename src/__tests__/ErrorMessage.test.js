import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ErrorMessage from '../components/ErrorMessage';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('ErrorMessage component', () => {
  test('renders with error', () => {
    const message = 'hello';
    const priority = 1;
    const tree = renderer
    .create(<ErrorMessage message={message} priority={priority} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('does not renders without error message', () => {
    const message = 'hello';
    const priority = 2;
    const tree = renderer
    .create(<ErrorMessage message={message} priority={priority} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});