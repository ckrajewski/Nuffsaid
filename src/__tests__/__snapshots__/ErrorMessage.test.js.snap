// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component does not renders without error message 1`] = `
<div
  className="makeStyles-container-2"
  style={
    Object {
      "visibility": "hidden",
    }
  }
>
  <div
    className="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation5 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root makeStyles-grid-5 MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      />
    </div>
  </div>
</div>
`;

exports[`ErrorMessage component renders with error 1`] = `
<div
  className="makeStyles-container-2"
  style={
    Object {
      "visibility": "hidden",
    }
  }
>
  <div
    className="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation5 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root makeStyles-grid-5 MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      />
    </div>
  </div>
</div>
`;
